### ISSUE TITLE
HotReload doesn't work with sass plugin

### ISSUE THREAD

**Issue by alexisvincent** [alexisvincent/systemjs-hot-reloader#37]
As far as I can see, the sass import tree isnt recognosed by systems module loading scheme as this is specific to sass. As such editing a child .scss file doesnt update the root. It might be a good idea for plugins to offer a hotReload hook so as to allow each plugin to handle its own hot reload logic, instead of adding something custom to the hot-reloader core.

Sass/css hot reloading is important, especially if this is going to integrated to the jspm core. @guybedford

**Comment by alexisvincent** [comment]
A hacky fix is to detect .scss and then manually delete the root .scss and reload it.

```js
hotReload (moduleName) {
    const self = this
    const start = new Date().getTime()
		
		if (moduleName.endsWith('.scss')) {
			var rootSCSS = 'app/web/app.scss!';
			return System.normalize(rootSCSS).then(function(normalizedName){
				System.delete(normalizedName)
				return System.import(rootSCSS)
			})
		}
				
    this.modulesJustDeleted = {}  // TODO use weakmap
    //etc...


```

**Comment by guybedford** [comment]
The way I expected to handle this was for the plugin itself to provide the hot reload hooks for the dummy module representing each sass file. But let me know if that approach doesn't work out?

**Comment by alexisvincent** [comment]
That should work. You would need to do an interception, similar to how I'm doing it here, as chakara will send over all the paths.

**Comment by npbenjohnson** [comment]
I looked into this a little, gave up and made a workaround. The plugin feeds in a hook to resolve @import statements inside of lib-sass, which means the plugin itself doesn't actually know what its children are.
For a workaround, I grabbed the locator hook from the sass plugin (so 'jspm:' style tags resolve), and put it in a node-sass task that builds my site css. Then just watch all sass files, build on change, and use browsersync's css reload on site.css .
